docs: adds CLI section in guide

luckylooke opened the pull request:
<!-- Thank you for contributing! -->

### Description

Adds CLI section to docs guide in same manner as [vitest does have](https://vitest.dev/guide/cli.html).

### Additional context

<!-- e.g. is there anything you'd like reviewers to focus on? -->
It was difficult to follow exact structure of the vitest document as it shares options between all commands. Here we have different options for each command, so I somehow tried to solve this.

Also I have notice type notations in options, vitest does not have such, but I consider them useful and kept them in shape `[type]`, not sure if it is OK
---

### What is the purpose of this pull request? <!-- (put an "X" next to an item) -->

- [ ] Bug fix
- [ ] New Feature
- [x] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/vitejs/vite/blob/main/.github/commit-convention.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

Commit:
docs: adds CLI section in guide

luckylooke commented:
Can you rerun task "CI / Build&Test: node-14, ubuntu-latest (pull_request)"? As I didn't touch anything related to it.. probably false alarm 🤔

patak-dev commented:
@luckylooke done, don't worry, there are some flaky tests we need to check out. Sorry for the noise.

patak-dev commented:
@luckylooke looks like there are some formatting issues with the table. See the preview here: https://deploy-preview-11025--vite-docs-main.netlify.app/guide/cli.html, was this looking ok for you locally? You can run `pnpm run docs` to check.

It would also be good to add a "CLI" entry to the sidebar after "Features" like we have in https://vitest.dev. You can add it [here](https://github.com/vitejs/vite/blob/fae0e234b92160bb6b6c14fe004647895459eded/docs/.vitepress/config.ts#L188)

luckylooke commented:
> @luckylooke looks like there are some formatting issues with the table. See the preview here: https://deploy-preview-11025--vite-docs-main.netlify.app/guide/cli.html, was this looking ok for you locally? You can run `pnpm run docs` to check.
> 
> It would also be good to add a "CLI" entry to the sidebar after "Features" like we have in https://vitest.dev. You can add it [here](https://github.com/vitejs/vite/blob/fae0e234b92160bb6b6c14fe004647895459eded/docs/.vitepress/config.ts#L188)

ouch.. it seems that some precommit task did broke source code of the file, because it is totaly changed and really broken, it looks fine before commit, I didn't expect that commit can break it.. I will investigate more

Commit:
fix: CR #11025

luckylooke commented:
Ok, problem was that I used bad escape characters `\\` instead of standard `\` which leads prettier to fail. But also I have found some extra places where I missed `|` which should be escaped.. so now double checked and should be all good. 

I have also added item into menu via config you suggested, thanks for catch.

And I have removed:
```md
---
title: Command Line Interface | Guide
---
```

part as it does not seem to have any efffect on output.. it was copypasted from vitest source, but I don't see reason having it there 🤷‍♂️ 

Thanks 🙏😊

patak-dev commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+
+| Options                  |                                                                 |
+| ------------------------ | --------------------------------------------------------------- |
+| `--host [host]`          | [string] specify hostname                                       |
```
What if we follow [Vite](https://vitest.dev/guide/cli.html) here so we can start each description capitalized, and we also use inline code for the types following our config guide?
```suggestion
| `--host [host]`          | Specify hostname (`string`)                                      |
```

luckylooke replied:
Ok, I will rewrite it in that manner 👍

Commit:
fix: CR - types in options tables #11025

luckylooke commented:
I did those changes, but for some reason I can't test it properly, still getting:
```cmd
pnpm run docs                                               

> vite-monorepo@ docs /Users/laky/Projects/vite
> vitepress dev docs

node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/laky/Projects/vite/node_modules/.pnpm/vitepress@1.0.0-draft.8/node_modules/vite/dist/node/index.js' imported from /Users/laky/Projects/vite/node_modules/.pnpm/vitepress@1.0.0-draft.8/node_modules/vitepress/dist/node/cli.js
Did you mean to import vite/packages/vite/index.cjs?
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```
I have tried reinstall node_modules, clear everything up, git clone project again and nothing helped, I can investigate more later.

patak-dev commented:
I tried it locally and it works fine, also the deployed version looks great: https://deploy-preview-11025--vite-docs-main.netlify.app/

Thanks for the changes!

sapphi-red commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+
+| Options                        |                                                                                                                     |
+| ------------------------------ | ------------------------------------------------------------------------------------------------------------------- |
+| `--target <target>`            | Transpile target (default: `'modules'`) (`string`)                                                                  |
```
```suggestion
| `--target <target>`            | Transpile target (default: `"modules"`) (`string`)                                                                  |
```

sapphi-red commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+| `--open [path]`          | Open browser on startup (`boolean \| string`)        |
+| `--outDir <dir>`         | Output directory (default: dist)(`string`)           |
+| `-c, --config <file>`    | Use specified config file (`string`)                 |
+| `--base <path>`          | Public base path (default: /) (`string`)             |
```
```suggestion
| `--base <path>`          | Public base path (default: `/`) (`string`)             |
```

sapphi-red commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+| `--strictPort`           | Exit if specified port is already in use (`boolean`) |
+| `--https`                | Use TLS + HTTP/2 (`boolean`)                         |
+| `--open [path]`          | Open browser on startup (`boolean \| string`)        |
+| `--outDir <dir>`         | Output directory (default: dist)(`string`)           |
```
```suggestion
| `--outDir <dir>`         | Output directory (default: `dist`)(`string`)           |
```

sapphi-red committed:
chore: tweaks

sapphi-red commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+| `--strictPort`           | Exit if specified port is already in use (`boolean`) |
+| `--https`                | Use TLS + HTTP/2 (`boolean`)                         |
+| `--open [path]`          | Open browser on startup (`boolean \| string`)        |
+| `--outDir <dir>`         | Output directory (default: `dist`)(`string`)           |
```
```suggestion
| `--outDir <dir>`         | Output directory (default: `dist`)(`string`)         |
```

sapphi-red commented on docs/guide/cli.md:
```diff
@@ -0,0 +1,126 @@
+| `--open [path]`          | Open browser on startup (`boolean \| string`)        |
+| `--outDir <dir>`         | Output directory (default: `dist`)(`string`)           |
+| `-c, --config <file>`    | Use specified config file (`string`)                 |
+| `--base <path>`          | Public base path (default: `/`) (`string`)             |
```
```suggestion
| `--base <path>`          | Public base path (default: `/`) (`string`)           |
```

sapphi-red committed:
chore: fix format

sapphi-red reviewed (approved):
Looks nice ✨

luckylooke commented:
Great :🎉, thanks 👏😊

patak-dev commented:
Thanks a lot @luckylooke ❤️
